tests/test_node.py

```diff
@@ -71,3 +72,46 @@ def test_run_as_root(
                     if record.levelno == logging.DEBUG:
                         found_expected_error = True
             assert found_expected_error
+
+    # An arbitrary time slice that is longer than it takes to spin up a cluster
+    # but avoids bash infinite loops.
+    @pytest.mark.timeout(60 * 15)
+    def test_popen_as_root(
+        self,
+        cluster_backend: ClusterBackend,
+        oss_artifact: Path,
+    ) -> None:
+        """
+        It is possible to run commands as root asynchronously
+        """
+        with Cluster(
+            agents=0,
+            public_agents=0,
+            cluster_backend=cluster_backend,
+            generate_config_path=oss_artifact,
+        ) as cluster:
+            (master, ) = cluster.masters
+            popen_1 = master.popen_as_root(
+                args=[
+                    '(mkfifo', '/tmp/pipe', '|', 'true)'
```

Comment:
The problem with the original test is that, `while` is not a construct available on all shells and thus our command would not run at all in a python subprocess. This could be remedied by setting `shell=True` for the `Popen` command. This lead to the next problem. The command we pass to `popen_as_root` via `args` is run via `SSH` which did not support running while loops, so I just dropped the idea.

Furthermore I think this idea is cleaner. It uses `named pipes` which are blocking mechanisms. If a process tries to read from a pipe it blocs until there is a process writing on it and vice versa, if a process writes to a pipe, it blocks until there is a process reading from it. In our case we have:
`popen_1: Create a pipe if not existing and read from it.`
`popen_2: Create a pipe if not existing and write to it.`
The commands could have been much simpler but we have to take care of situations where the processes do not start in the specified order.